csrc/quantization/machete/machete_mm_launcher.cuh

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <torch/all.h>
+#include <Python.h>
+
+#include "machete_mm_kernel.cuh"
+#include "cutlass_extensions/torch_utils.hpp"
+
+namespace machete {
+
+struct PytorchArguments {
+  torch::Tensor const A;
+  torch::Tensor const B;
+  c10::optional<torch::Tensor> const& scales;
+  c10::optional<torch::Tensor> const& zeros;
+  c10::optional<int64_t> group_size;
+  c10::optional<torch::Tensor> const& C;
+  c10::optional<double> alpha;
+  c10::optional<double> beta;
+  c10::optional<std::string> schedule;
+};
+
+template <typename KernelSpeacialization>
```

Comment:
Also, what do you think about a slightly different paradigm where instead of passing the type of the specialization, you passed a type that contained the arguments for it and then passed that to a `using KernelTraits = KernelSpecializationTraits<KernelSpecialization>;`, as then the IDE could help the user navigate to the proper place in the code where specialization is defined. It would also make code completion better and errors appear earlier.

Reply:
Im not sure I understand what you are suggesting here